app/javascript/components/student/editor/Settings.tsx

```diff
@@ -0,0 +1,109 @@
+import React, { useState, useRef, useEffect } from 'react'
+import { usePopper } from 'react-popper'
+import { Icon } from '../../common/Icon'
+import { Keybindings } from './types'
+
+export function Settings({
+  theme,
+  keybindings,
+  wrap,
+  setTheme,
+  setKeybindings,
+  setWrap,
+}: {
+  theme: string
+  keybindings: Keybindings
+  wrap: 'off' | 'on' | 'wordWrapColumn' | 'bounded'
+  setTheme: (theme: string) => void
+  setKeybindings: (keybinding: Keybindings) => void
+  setWrap: (wrap: 'off' | 'on' | 'wordWrapColumn' | 'bounded') => void
+}) {
+  const [open, setOpen] = useState(false)
+  const buttonRef = useRef<HTMLButtonElement>(null)
+  const panelRef = useRef<HTMLDivElement>(null)
+  const componentRef = useRef<HTMLDivElement>(null)
+  const { styles, attributes, update } = usePopper(
+    buttonRef.current,
+    panelRef.current,
+    {
+      placement: 'bottom',
+    }
+  )
+
+  const handleClickOutside = (e: MouseEvent) => {
+    if (componentRef.current?.contains(e.target as Node)) {
+      return
+    }
+
+    setOpen(false)
+  }
+
+  useEffect(() => {
```

Comment:
This is to detect whether the user has clicked outside the panel, and if they do, we close the panel.

Reply:
If you need a comment here in the review, you probably want to place the comment on line 40 (It's a good comment to have)